### PR TITLE
Refactored inference API, and extracted prompt template as individual module

### PR DESCRIPTION
**Pull request by XinyuYe-Intel**
## Type of Change

API refactor
API not changed

## Description

Refactored inference API, and extracted prompt template as individual module.
JIRA ticket: xxx

## Expected Behavior & Potential Risk

pass UT

## How has this PR been tested?

UT

## Dependency Change?

no library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/utils/mpt_trace.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    with torch.inference_mode():
+        traced_model = torch.jit.trace(model, example_inputs=(inputs["input_ids"], inputs["past_key_values"],
+                                       inputs["attention_mask"]), strict=False)
+        traced_model = torch.jit.freeze(traced_model.eval())
+        traced_model(**inputs)
+        traced_model(**inputs)
+    return traced_model
+
+
+class MPTTSModelForCausalLM(TSModelForCausalLM):
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/utils/mpt_trace.py:46]: should this be under llm?

**Comment by hshen14** [comment]
@XinyuYe-Intel  please check the code coverage issue. @VincyZhang

File: intel_extension_for_transformers/neural_chat/utils/mpt_trace.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from typing import Optional, Tuple
+from transformers.modeling_outputs import CausalLMOutputWithPast
+from optimum.intel.generation.modeling import TSModelForCausalLM # pylint: disable=E0401
+
+
+def jit_trace_mpt_7b(model):
```
Comment (changwangss) [inline intel_extension_for_transformers/neural_chat/utils/mpt_trace.py:24]: Could you make this function more generalizable  as the utility? for example for ModelForCausalLM.

File: intel_extension_for_transformers/neural_chat/utils/mpt_trace.py (hunk 1 above)
Comment (XinyuYe-Intel) [reply intel_extension_for_transformers/neural_chat/utils/mpt_trace.py]: done.

File: intel_extension_for_transformers/neural_chat/utils/model_utils.py [hunk 3]
```diff
@@ -334,25 +124,33 @@ def import_deepspeed():
...
-            " git+https://github.com/HabanaAI/DeepSpeed.git@1.10.0`."
+            " git+https://github.com/HabanaAI/DeepSpeed.git@1.11.0`."
         )
     # Initialize process(es) for DeepSpeed
     deepspeed.init_distributed(dist_backend="hccl")
     logger.info("DeepSpeed is enabled.")
 
 
-def init_deepspeed_inference(model, model_name_or_path, use_hpu_graphs):
+def init_deepspeed_inference(model, model_name_or_path, use_hpu_graphs, is_meta):
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/utils/model_utils.py:134]: why need this?

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
 from intel_extension_for_transformers.neural_chat.pipeline.plugins.prompt.prompt_template \
     import generate_intent_prompt
-from intel_extension_for_transformers.llm.inference import predict
+from intel_extension_for_transformers.neural_chat.utils.model_utils import predict
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py:21]: predict/predict_stream is better in inference?

File: intel_extension_for_transformers/neural_chat/utils/model_utils.py (hunk 3 above)
Comment (XinyuYe-Intel) [reply intel_extension_for_transformers/neural_chat/utils/model_utils.py:134]: it's from WangYi's change on workflow code.

Comment (hshen14) [reply intel_extension_for_transformers/neural_chat/utils/model_utils.py:134]: looks not good. can we fix? if not easy, can be in next pr

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py (hunk 4 above)
Comment (XinyuYe-Intel) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py:21]: is changing model_utils to inference_utils ok?

File: intel_extension_for_transformers/neural_chat/utils/prompt.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (XinyuYe-Intel) [reply intel_extension_for_transformers/neural_chat/utils/prompt.py]: done.

**Review by hshen14 — APPROVED** [review]

**Comment by hshen14** [comment]
@XinyuYe-Intel  are we aligned with prompt design/organization?

**Comment by XinyuYe-Intel** [comment]
> @XinyuYe-Intel are we aligned with prompt design/organization?

yes, I'll create another PR for prompt.

**Pull request merged**